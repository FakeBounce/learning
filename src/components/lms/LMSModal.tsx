import { styled } from '@mui/material/styles';
import Dialog from '@mui/material/Dialog';
import DialogTitle from '@mui/material/DialogTitle';
import DialogContent from '@mui/material/DialogContent';
import DialogActions from '@mui/material/DialogActions';
import IconButton from '@mui/material/IconButton';
import Iconify from '@src/components/iconify/Iconify';
import ActionButton from '@src/components/lms/ActionButton';
import { Trans } from '@lingui/macro';
import { ReactNode } from 'react';

const StyledDialog = styled(Dialog)(({ theme }) => ({
  '& .MuiDialogContent-root': {
    padding: theme.spacing(2)
  },
  '& .MuiDialogActions-root': {
    padding: theme.spacing(1)
  },
  '& .MuiPaper-root': {
    borderRadius: theme.shape.customBorderRadius.extraLarge,
    padding: theme.spacing(1)
  }
}));

interface LMSModalProps {
  title: ReactNode;
  children: ReactNode;
  open: boolean;
  cancelAction?: () => void;
  validateAction?: () => void;
  onClose: () => void;
  isLoading?: boolean;
}

export default function LMSModal({
  title,
  children,
  open,
  onClose,
  validateAction,
  cancelAction,
  isLoading = false
}: LMSModalProps) {
  return (
    <StyledDialog onClose={onClose} aria-labelledby="customized-dialog-title" open={open}>
      <DialogTitle sx={{ m: 0, p: 2 }}>{title}</DialogTitle>
      <IconButton
        aria-label="close"
        onClick={onClose}
        sx={{
          position: 'absolute',
          right: 8,
          top: 8,
          color: (theme) => theme.palette.grey[500]
        }}
      >
        <Iconify icon="material-symbols:close" />
      </IconButton>
      <DialogContent sx={{ display: 'flex', flexDirection: 'column', gap: 2 }}>
        {children}
      </DialogContent>
      <DialogActions sx={{ gap: 1 }}>
        {cancelAction && (
          <ActionButton actionType="cancel" onClick={cancelAction} loading={isLoading}>
            <Trans>Annuler</Trans>
          </ActionButton>
        )}
        {validateAction && (
          <ActionButton onClick={validateAction} loading={isLoading}>
            <Trans>Valider</Trans>
          </ActionButton>
        )}
      </DialogActions>
    </StyledDialog>
  );
}
