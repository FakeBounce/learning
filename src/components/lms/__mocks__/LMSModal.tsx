import Iconify from '@src/components/iconify/Iconify';
import { Trans } from '@lingui/macro';
import { ReactNode } from 'react';

interface LMSModalProps {
  title: ReactNode;
  children: ReactNode;
  open: boolean;
  cancelAction?: () => void;
  validateAction?: () => void;
  onClose: () => void;
  isLoading?: boolean;
}

export default function LMSModal({
  title,
  children,
  open,
  onClose,
  validateAction,
  cancelAction,
  isLoading = false
}: LMSModalProps) {
  if (!open) {
    return null;
  }
  return (
    <div>
      <div>{title}</div>
      <button aria-label="close" onClick={onClose}>
        <Iconify icon="material-symbols:close" />
      </button>
      <div>{children}</div>
      <div>
        {cancelAction && (
          <button onClick={cancelAction} disabled={isLoading}>
            <Trans>Annuler</Trans>
          </button>
        )}
        {validateAction && (
          <button onClick={validateAction} type="submit" role="submit" disabled={isLoading}>
            <Trans>Valider</Trans>
          </button>
        )}
      </div>
    </div>
  );
}
