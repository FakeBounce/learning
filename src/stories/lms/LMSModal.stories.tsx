import type { Meta, StoryFn } from '@storybook/react';
import LMSModal from '@src/components/lms/LMSModal';
import { useState } from 'react';

const meta: Meta = {
  title: 'Example/LMSModal',
  component: LMSModal,
  tags: ['autodocs'],
  parameters: {
    layout: 'centered'
  }
};

export default meta;

const Template: StoryFn<typeof LMSModal> = (args: any) => {
  const [open, setOpen] = useState(false);

  const handleClose = () => {
    setOpen(false);
  };

  const handleOpen = () => {
    setOpen(true);
  };

  return (
    <>
      <button onClick={handleOpen}>Open modal</button>
      <LMSModal {...args} open={open} onClose={handleClose} />
    </>
  );
};

export const Default = Template.bind({});
Default.args = {
  title: 'Modal Title',
  children: 'Modal Content',
  cancelAction: () => console.log('Cancelled'),
  validateAction: () => console.log('Validated')
};
