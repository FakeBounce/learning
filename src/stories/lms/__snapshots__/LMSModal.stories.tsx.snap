// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example/LMSModal Default smoke-test 1`] = `
<button>
  Open modal
</button>
`;
